doc/decisions/1_in_discussion/spec_expressiveness.md

```diff
@@ -2,22 +2,61 @@
 
 ## Problem
 
-Currently, you can easily come into wrong assumptions that something would work in the specification.
+The current implementation of the spec plugin is copying metadata to all other namespaces.
+Metadata e.g. `type` or `description` describes a certain key.
 
-We need to find minimal requirements to implement a sane spec plugin.
+Sometimes there are keys with similar or identical specifications.
+We want to define those in a single place and not have to manually copy metadata.
+`Globbing` expressions in key names comes into place at that time. 
+By providing certain wildcard characters e.g. `_` or `#` one can match multiple keys.
+This removes the need to explicitly define specification for keys.
+
+To describe the use case where `globbing` expressions make sense, we will use a REST application.
+This REST application uses multiple services which the user is not aware of at time of configuration.
+It could use a database, message broker, etc. but this is not yet known.
+A solution to this is `globbing` i.e. using wildcards in key names.
+The thing to be aware of is to define configuration keys which are necessary for every service.
+In this example we use `port` and `url`.
+
+The parent key for our specification is `/sw/org/app-name/#0/development`.
+
+```ni
+[service/_/port]
+meta:/require = true
+meta:/default = 8080
+meta:/description = "The port of the service."
+
+[service/_/url]
+meta:/require = true
+meta:/description = "The url of the service."
+```
+The `_` would match every `service` we are using in our application e.g. `database`, `message broker`, etc.
```

Comment:
If I want to explain the need for globbing IMO it is best to show that it is worse (or impossible) without globbing. In your example you could start like this:

````markdown
If we have `database`, `broker` and `mailer` services we could use the specification:

```ni
[service/database/port]
meta:/require = true
meta:/default = 8080
meta:/description = "The port of the database service."

[service/database/url]
meta:/require = true
meta:/description = "The url of the database service."

[service/broker/port]
meta:/require = true
meta:/default = 8080
meta:/description = "The port of the broker service."

[service/broker/url]
meta:/require = true
meta:/description = "The url of the broker service."

[service/mailer/port]
meta:/require = true
meta:/default = 8080
meta:/description = "The port of the mailer service."

[service/mailer/url]
meta:/require = true
meta:/description = "The url of the mailer service."
```

This is already very repetitive and error prone. But writing the specification becomes completely when the set of services is not known in advance. To solve this we need some way to allow for dynamic/unknown parts in the keyname. That is what we call Globbing.

[...]
````

Reply:
Thank you, this is much better.